### PR TITLE
fix(toolbar): Don't try to load hedgehog config without auth

### PR DESCRIPTION
**Pull request by Twixes**
## Problem

This toast shows up when Toolbar is unauthenticated:

Screenshot 2024-10-01 at 20 11 14

## Changes

Toast should be no more.

### CODE REVIEW

File: frontend/src/lib/components/HedgehogBuddy/hedgehogBuddyLogic.ts [hunk 1]
```diff
@@ -54,7 +54,11 @@ export const hedgehogBuddyLogic = kea<hedgehogBuddyLogicType>([
...
                     const endpoint = '/api/users/@me/hedgehog_config'
-                    if (toolbarConfigLogic?.findMounted()?.values.temporaryToken) {
+                    const maybeMountedToolbarConfigLogic = toolbarConfigLogic.findMounted()
+                    if (maybeMountedToolbarConfigLogic) {
+                        if (!maybeMountedToolbarConfigLogic.values.isAuthenticated) {
+                            return null
+                        }
                         return await (await toolbarFetch(endpoint, 'GET')).json()
                     }
                     return await api.get<Partial<HedgehogConfig>>(endpoint)
```
Comment (pauldambra) [inline frontend/src/lib/components/HedgehogBuddy/hedgehogBuddyLogic.ts:64]: i find this hard to parse...

```
 const isInTheToolbar = toolbarConfigLogic.findMounted()
                    if (isInTheToolbar) {
                        if (isInTheToolbar.values.isAuthenticated) {
                            return await (await toolbarFetch(endpoint, 'GET')).json()
                        }
                        return null
                    }
```

maybe that's slightly easier but not hugely 🤷

**Review by pauldambra — APPROVED** [review]

**Review by benjackwhite — APPROVED** [review]

File: frontend/src/lib/components/HedgehogBuddy/hedgehogBuddyLogic.ts (hunk 1 above)
Comment (Twixes) [reply frontend/src/lib/components/HedgehogBuddy/hedgehogBuddyLogic.ts]: I see what you mean. Although also not sure about `isInTheToolbar`, as that looks much like a bool. Hopefully this is  also better: [`775a336` (#25323)](https://github.com/PostHog/posthog/pull/25323/commits/775a3364cc08269322192678d7995e158e76517f)

Comment (pauldambra) [reply frontend/src/lib/components/HedgehogBuddy/hedgehogBuddyLogic.ts]: > also not sure about isInTheToolbar, as that looks much like a bool

fair! 

nice compromise :)
